EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -141,9 +111,15 @@ protected List<int> GetBinsToSearch(Ms2ScanWithSpecificMass scan, List<int>[] Fr
                     //convert to an int since we're in discreet 1.0005...
                     int fragmentBin = (int)(Math.Round(masses[i].ToMass(1) / 1.0005079) * 1.0005079 * FragmentBinsPerDalton);
 
-                    if (FragmentIndex[fragmentBin] != null)
+                    List<int> bin = FragmentIndex[fragmentBin];
```

Comment:
If my understanding is right. This here is same as "List<int> binsToSearch = new List<int>();" in the old way.

Reply:
`List binsToSearch = new List();` creates a new list, `List<int> bin = FragmentIndex[fragmentBin];` creates a new reference to an existing list